static/js/stream_edit.js

```diff
@@ -63,6 +63,51 @@ exports.get_users_from_subscribers = function (subscribers) {
     });
 };
 
+exports.get_retention_policy_text_for_subscription_type = function (sub) {
+    let message_retention_days = sub.message_retention_days;
+    if (page_params.realm_message_retention_days === -1 && (sub.message_retention_days === null ||
+            sub.message_retention_days === -1)) {
+        return;
+    }
+
+    if (sub.message_retention_days === -1) {
+        return i18n.t("Messages in this stream will be retained forever.");
+    }
+
+    if (message_retention_days === null)  {
+        message_retention_days = page_params.realm_message_retention_days;
+    }
```

Comment:
I think you can write this more cleanly by having this be the first thing you do.  then that first if statement doesn't need to have so much complexity.

Reply:
@timabbott  I am not sure that it will become clean because, in that case also we want to check for if stream value is set to -1 and in this case also both organization-level and stream-level setting will be forever and we do not want to show any text.

I also have missed the case of `page_params.realm_message_retention_days` being null here. I was thinking of something like this for the case of not showing any text:
`[-1, null].includes(page_params.realm_message_retention_days) && [-1, null].includes(sub.message_retention_days)`
I have seen this being used in other places, is it fine to use this?